specs/ContextMenuRequested.md

```diff
@@ -0,0 +1,764 @@
+        /// and the `args` parameter is set to `null`.
+        HRESULT Invoke(
+            [in] ICoreWebView2ContextMenuItem* sender, [in] IUnknown* args);
+    }
+
+    /// Event args for the `ContextMenuRequested` event. Will contain the selection information 
+    /// and a collection of all of the default context menu items that the WebView
+    /// would show. Allows the app to draw its own context menu or add/remove 
+    /// from the default context menu.
+    [uuid(a1d309ee-c03f-11eb-8529-0242ac130003), object, pointer_default(unique)]
+    interface ICoreWebView2ContextMenuRequestedEventArgs : IUnknown
+    {
+        /// Gets the collection of `ContextMenuItem` objects. 
+        /// See `ICoreWebView2ContextMenuItemCollection` for more details.
+        [propget] HRESULT MenuItems([out, retval] ICoreWebView2ContextMenuItemCollection** value);
+
+        /// Gets the information associated with the requested context menu. 
+        /// See `ICoreWebView2ContextMenuInfo` for more details.
+        [propget] HRESULT ContextMenuInfo([out, retval] ICoreWebView2ContextMenuInfo** value);
+
+        /// Sets the selected command for the WebView to execute. The value is
+        /// obtained via the `ContextMenuItem` CommandId property
+        /// The default value is -1 which means that no selected occured.
+        [propput] HRESULT SelectedCommandId([in] INT32 value);
+
+        /// Gets the selected CommandId.
+        [propget] HRESULT SelectedCommandId([out, retval] INT32* value);
+
+        /// Sets whether the `ContextMenuRequested` event is handled by host after
+        /// the event handler completes or if there is a deferral then after the deferral is completed.  
+        /// If Handled is set to TRUE then WebView2 will not display a context menu and will instead
+        /// use the SelectedCommandId property to indicate which, if any, context menu item to invoke.
+        /// If after the event handler or deferral completes Handled is set to FALSE then WebView2 
+        /// will display a context menu based on the contents of the MenuItems property. 
+        /// The default value is FALSE.
+        [propput] HRESULT Handled([in] BOOL value);
+
+        /// Gets whether the `ContextMenuRequested` event is handled by host.
+        [propget] HRESULT Handled([out, retval] BOOL* value);
+
+        /// Returns an `ICoreWebView2Deferral` object. Use this operation to
+        /// complete the event when the custom context menu is closed.
+        HRESULT GetDeferral([out, retval] ICoreWebView2Deferral** deferral);
+    }
+
+    /// Represents the information regarding the context menu. Includes the location of the request, 
+    /// the context selected and the appropriate data used for the actions of a context menu.
+    [uuid(b8611d99-eed6-4f3f-902c-a198502ad472), object, pointer_default(unique)]
+    interface ICoreWebView2ContextMenuInfo : IUnknown
```

Comment:
Is there a particular reason why this is broken out into a sub-object rather than just being additional properties on the EventArgs?

Reply:
See next related question